lib/core/versions/latest/CoreProofFile.ts

```diff
@@ -32,4 +52,100 @@ export default class CoreProofFile {
 
     return compressedRawData;
   }
+
+  /**
+   * Parses and validates the given core proof file buffer.
+   * @param coreProofFileBuffer Compressed core proof file.
+   * @throws `SidetreeError` if failed parsing or validation.
+   */
+  public static async parse (coreProofFileBuffer: Buffer, expectedDeactivatedDidUniqueSuffixes: string[]): Promise<CoreProofFile> {
+    let coreProofFileDecompressedBuffer;
+    try {
+      const maxAllowedDecompressedSizeInBytes = ProtocolParameters.maxProofFileSizeInBytes * Compressor.estimatedDecompressionMultiplier;
+      coreProofFileDecompressedBuffer = await Compressor.decompress(coreProofFileBuffer, maxAllowedDecompressedSizeInBytes);
+    } catch (error) {
+      throw SidetreeError.createFromError(ErrorCode.CoreProofFileDecompressionFailure, error);
+    }
+
+    let coreProofFileModel;
+    try {
+      coreProofFileModel = await JsonAsync.parse(coreProofFileDecompressedBuffer);
+    } catch (error) {
+      throw SidetreeError.createFromError(ErrorCode.CoreProofFileNotJson, error);
+    }
+
+    if (coreProofFileModel.operations === undefined) {
+      throw new SidetreeError(ErrorCode.CoreProofFileOperationsNotFound, `Core proof file does not have any operation proofs.`);
+    }
+
+    const operations = coreProofFileModel.operations;
+    const allowedProperties = new Set(['recover', 'deactivate']);
```

Comment:
Can't this just use the validateObjectContainsOnlyAllowedProperties function you created?

Reply:
Yes it can, I added the lib method after this and was being lazy, was hoping to do a full codebase sweep separately, but will do this one now since you asked!